### DISCUSSION TITLE
How to download a file !

### DISCUSSION THREAD

**Discussion by Abiologist** [community/community#52607] [New to GitHub]
Dear All,

Just in case someone starting in Github wants to download a file - the flow is:

Right-click on "Raw" - "Save Link As"

Perhaps this is especially designed to discourage someone from downloading something (?)  - because I sure can't think of a reason why I would want to click on "Save Link As" when I don't want to save a link - I want to download a file !! What's wrong with the word "Download" anyway ?

Yours sincerely

Jeremy Clark

**Comment by CodeMaster7000** [comment]
Hi @Abiologist,

You can download an image/GIF file by going to the file you want to click on, clicking "Download" and then "save image as". An example is shown below:

![Screenshot 2023-04-12 104349](https://user-images.githubusercontent.com/95772109/231420286-71acd582-b27e-4b3c-9f2b-33669b1b27e8.png)

As for whole repositories, click on "<>Code" and "Download ZIP" to obtain the files you want:

![Screenshot 2023-04-12 104714](https://user-images.githubusercontent.com/95772109/231421171-6f0fe643-2508-4466-a6c2-7d379e7b7061.png)

Hope this helps and have a nice day!

**Comment by Abiologist** [comment]
Dear CodeMaster7000,

This is only correct for some types of file.
I've just checked a .txt file - I cannot follow the instructions you gave.
1. Either I am on the site where Code, Readme and About are available -
plus the list of files - in which case I cannot see an option to download.
If I click on the individual file - it just brings me to the page where I
can see all of the text in the file - with no option to download.
2. Or - I click on "Go to" file - this gives me a list of files in blue -
again with no option to download. If I click on the individual file it just
brings me to the page where I can see all of the text in the file - with no
option to download.
3. The only way to download this file is to right-click on "raw" and then
click "Save Link As" - which is of course ridiculous.

Yours sincerely,

Jeremy Clark






On Wed, Apr 12, 2023 at 11:44 AM CodeMaster7000 ***@***.***>
wrote:

> Hi @Abiologist <https://github.com/Abiologist>,
>
> You can also download a file by going to the file you want to click on,
> clicking "Download" and then "save image as".
>
> [image: Screenshot 2023-04-12 104349]
> <https://user-images.githubusercontent.com/95772109/231420286-71acd582-b27e-4b3c-9f2b-33669b1b27e8.png>
>
> Hope this helps and have a nice day!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/community/discussions/52607#discussioncomment-5590596>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADYLLF7NLJI4YFKABAMVRITXAZ2Q7ANCNFSM6AAAAAAW3NF63I>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

dr hab. n. med. Jeremy Clark, prof. PUM,

Department of Clinical & Molecular Biochemistry,

Pomeranian Medical University in Szczecin; Pomorski Uniwersytet Medyczny w
Szczecinie,

Zakład Biochemii Klinicznej i Molekularnej,

ul. Powstancow Wlkp. 72, 70-111 Szczecin, Poland.

Tel. 0048 570112192; Tel. 004891 4661490; Fax. 004891 4661492

Email:   ***@***.***

**Comment by CodeMaster7000** [comment]
@Abiologist The option to download "Code" is on the repository's main page.

However, after some research, I have found that to download an individual file of formats like .txt, the only way to download them is in the way you told me.
